### ISSUE TITLE
Incorrect date warning in l3dvipdfmx

### ISSUE THREAD

**Issue by blefloch** [latex3/latex3#509]
When running platex on the following file that requires `expl3` from after a certain date (the precise date is not important) I get a warning that `l3dvipdfmx` is not new enough.

    \RequirePackage{expl3}[2018/02/21]
    \stop

gives

    LaTeX Warning: You have requested, on input line 283, version
               `2018/02/21' of package l3dvipdfmx,
               but only version
               `'
               is available.

It seems that the minimum date for `expl3.sty` is kept when loading `l3dvipdfmx.def`, while the definition of `\ProvidesExplFile` in `expl3.sty` does not pass its date argument back to LaTeX2e's data structures (such as by setting `ver@l3dvipdfmx.def`.

**Comment by u-fischer** [comment]
I can't reproduce this with the expl3 from texlive.  Are you using the development code?

**Comment by blefloch** [comment]
Yes, I'm using the development code.  Let me try with the TeXLive one.

**Comment by blefloch** [comment]
Same result.  Did you use pLaTeX?  Here are my TeX file and log file.

    \listfiles
    \RequirePackage{expl3}[2018/02/21]
    \stop

[test.log](https://github.com/latex3/latex3/files/2714566/test.log)

**Comment by u-fischer** [comment]
My latex is newer. You have a format `based on LaTeX2e <2018-04-01>`.

**Comment by blefloch** [comment]
Hum... I'm using stock TeXLive there (updated this morning).

**Comment by u-fischer** [comment]
Well the platex format was created on 12-13, perhaps an older latex was active/found at this moment. What happens if you run fmtutil-sys for ptex?
